src/Compilers/Core/Portable/Text/CompositeText.cs

```diff
@@ -288,13 +288,10 @@ private static void CombineSegments(ArrayBuilder<SourceText> segments, int segme
 
                     // count how many contiguous segments are reducible
                     int count = 1;
-                    for (int j = i + 1; j < segments.Count; j++)
+                    for (int j = i + 1; j < segments.Count && segments[j].Length <= segmentSize; j++)
```

Comment:
From https://github.com/dotnet/roslyn/pull/69359#pullrequestreview-1561266760:

> this has different semantics from before. previously it would skip over segments less than the segment size, now it stops looping once it hits one with a length less than the segment size. is that ok?

> Right. I believe this is more correct because below, the code combining the segments only combines contiguous segments and doesn't skip over any, so it seems like this is currently a bug. I discovered this issue thanks to the newly added assertions for length in `StringTextWriter` and `LargeTextWriter` - a unit test was failing until I changed this to only count contiguous segments.

Reply:
Also, the comment above literally says "contiguous segments" 😄 so it seems like this was an oversight.